nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1340 @@
+        mim.series_exponent = 0
+        mim.series_start = self.start
+        mim.series_step = self.step
+        mim.number_of_series_points = self.size
+        mim.series_unit = self.unit
+        return mim
+
+    _unit = None
+
+    @property
+    def unit(self):
+        return self._unit
+
+    @unit.setter
+    def unit(self, value):
+        if value.upper() not in ("SECOND", "HERTZ", "METER", "RADIAN"):
+            raise ValueError("Series unit should be one of " +
+                             "('second', 'hertz', 'meter', or 'radian'")
+        self._unit = value.upper()
+
+    def extend(self, other_axis):
+        """
+        Concatenates two Series
+
+        Note: this will ignore the start point of the other axis
+
+        Parameters
+        ----------
+        other_axis : Series
+            other axis
+
+        Returns
+        -------
+        Series
+        """
+        if other_axis.step != self.step:
+            raise ValueError('Can only concatenate Series with the same step size')
+        if other_axis.unit != self.unit:
+            raise ValueError('Can only concatenate Series with the same unit')
+        return Series(self.start, self.step, self.size + other_axis.size, self.unit)
+
+    def __len__(self):
+        return self.size
+
+    def __eq__(self, other):
+        """
+        True if start, step, size, and unit are the same.
+        """
+        return (
+                isinstance(other, Series) and
+                self.start == other.start and
+                self.step == other.step and
+                self.size == other.size and
+                self.unit == other.unit
+        )
+
+    def __add__(self, other):
+        """
+        Concatenates two Series
+
+        Parameters
+        ----------
+        other : Series
+            Time Series to append at the end of the current time Series.
+            Note that the starting time of the other time Series is ignored.
+
+        Returns
+        -------
+        Series
+            New time Series with the concatenation of the two
+
+        Raises
+        ------
+        ValueError
+            raised if the repetition time of the two time Series is different
+        """
+        if isinstance(other, Series):
+            return self.extend(other)
+        return NotImplemented
+
+    def __getitem__(self, item):
+        if isinstance(item, slice):
+            step = 1 if item.step is None else item.step
+            idx_start = ((self.size - 1 if step < 0 else 0)
+                         if item.start is None else
+                         (item.start if item.start >= 0 else self.size + item.start))
+            idx_end = ((-1 if step < 0 else self.size)
+                       if item.stop is None else
+                       (item.stop if item.stop >= 0 else self.size + item.stop))
+            if idx_start > self.size:
+                idx_start = self.size - 1
```

Comment:
In lists:

```Python
>>> a = list(range(5))
>>> a[6:]
[]
```

By the current logic, this should be equivalent to `a[4:]`, or `[4]`. So I think if we want consistency with lists:

```suggestion
                idx_start = self.size
```

Reply:
These lines are meant to deal with the following situation:
>>> a = list(range(5))
>>> a[6::-1]
[4, 3, 2, 1, 0]

In that case the `idx_start` is `self.size - 1`. I will alter the if-statement to by `if idx_start > self.size and step < 0` to clarify this.